src/protocol/modulus_conversion/convert_shares.rs

```diff
@@ -108,6 +108,37 @@ impl ConvertShares {
     }
 }
 
+pub async fn convert_all_shares<F: Field>(
+    ctx: &ProtocolContext<'_, F>,
+    input: &[u64],
+    num_bits: u8,
+) -> Result<Vec<Vec<Replicated<F>>>, BoxError> {
+    let converted_shares = try_join_all(zip(repeat(ctx), input).enumerate().map(
+        |(record_id, (ctx, row))| async move {
+            ConvertShares::new(XorShares {
+                num_bits,
+                packed_bits: *row,
+            })
+            .execute(
+                ctx.narrow(&ModulusConversion(record_id)),
+                RecordId::from(record_id),
+            )
+            .await
+        },
+    ))
+    .await?;
+
+    //Transpose the output to return bit-wise results
+    let cols = vec![Replicated::new(F::ZERO, F::ZERO); converted_shares.len()];
```

Comment:
An allocation on this scale is not ideal.  I would instead use std::iter::repeat()

Reply:
not needed anymore